pinot-core/src/main/java/org/apache/pinot/core/operator/query/AggregationGroupByOrderByOperator.java

```diff
@@ -145,4 +148,19 @@ public ExecutionStatistics getExecutionStatistics() {
     return new ExecutionStatistics(_numDocsScanned, numEntriesScannedInFilter, numEntriesScannedPostFilter,
         _numTotalDocs);
   }
+
+  private int calculateMinTrimSize () {
+    // In query option, if a positive min trim size is given, we use it to override the server settings. Otherwise
+    // check if a simple boolean option is given and use default trim size.
+    QueryOptions queryOptions =  new QueryOptions(_queryContext.getQueryOptions());
+    Boolean queryOptionEnableTrim = queryOptions.isEnableSegmentTrim();
+    Integer queryOptionTrimSize = queryOptions.getMinSegmentTrimSize();
+    if (queryOptionTrimSize != null && queryOptionTrimSize > 0) {
+      return queryOptionTrimSize;
+    } else if (queryOptionEnableTrim != null && queryOptionEnableTrim) {
+      // TODO: if query option is true and a server minTrimSize is given, shall we use default value or server config?
```

Comment:
Here if `_minSegmentTrimSize` is configured (> 0), we should use the server config. If not, use the default